### ISSUE TITLE
Ridgeline Plot Troubles

### ISSUE THREAD

**Issue by rami8797** [earthlab-education/ea-bug-queue#3]
For: (@eculler )
From: (@rami8797)

## The Project
I'm trying to make a ridgeline plot using reflectance data from a NEON h5 file.

## The Issue
I'm unable to make said plot. 

- [x] **The Goal**
I should be making a ridgeline plot that shows the differences of h5 band reflectance densities.

- [x] **The Problem**
The entire notebook takes about 30 minutes to run. There are a couple of cells at the end of the notebook that take about 10 minutes each. I have one code cell commented out because I have let it run for close to an hour with no success. The figure outputs of the cells that do run don't look correct. Each band's reflectance density looks very similar, if not the same.

  We may be able to solve your problem from this information alone. You are also welcome and encouraged to comment on your classmates’ issues, which will count towards participation credit.

- [x] **Prior Attempts**
I have tried using matplotlib, seaborn, and joyplot from joypy to plot this figure. Joyplot is the fastest. However, all attempts yeield the same result (densities look the same). The stylistic outcomes of the matplotlib and seaborn attempts aren't what I want. I'd like the densities to be slightly overlapping, which joyplot easily achieves. Eventually, I'd like the entire figure orientation to be shifted so that the densities are vertical instead of horizontal, but one thing at a time, I suppose.

- [x] **System Info**
Sometimes problems are related to:
  - [x] Operating system and version (Windows 10)
  - [x] RAM (Total Physical: 12.0 GB, Available Physical Memory: 5.73 GB)
  - [x] Number of Cores (2)
  - [x] Number of Logistical Processors (4)
  - [x] available disk space (791/916 GB)

**Comment by eculler** [comment]
@rami8797 I agree this plot should not take that long on your system! I'll see if I can figure out why.

**Comment by eculler** [comment]
@rami8797 I went through the code, and I actually think  all those ways of making the plot are working (including joyplot with the modified syntax). It's just that the distributions are indeed very similar. Take a look at this plot (the kdes are colored by box - within the box each are very similar):
![image](https://user-images.githubusercontent.com/3465768/170319651-25e3d41c-dea8-4e75-b30d-76fb88cb8fd0.png)

and this ridgeline plot, again including wavelengths from all boxes:

![image](https://user-images.githubusercontent.com/3465768/170331986-415a8c85-1ee7-48d6-8bb6-be57d5b1b4a3.png)

I will upload my code so you can take a look at how I generated the plots, but I think the main issue here is that the amount of variation you have is pretty small.

**Comment by rami8797** [comment]
@eculler thanks for taking a look at it! It's good to know it wasn't a coding problem! Looking at your code would be very helpful as well.

**Comment by eculler** [comment]
@rami8797 I thought I would get to cleaning it up sooner, so for now I uploaded commit f1aa99a which contains a *-response.ipynb file with my code. There's not a ton of explanation, so write back here if something doesn't make sense. There's examples of plotting, as well as some functions that help to use pandas in what I think is a cleaner way.
